### PR TITLE
Code Enhancements

### PR DESCRIPTION
**Pull request by WeihanLi**
- Keep private members readonly in `StringTokenProvider`
- Use `long.TryParse` instead of `int.TryParse`
- Sealed internal types(maybe we could enforce this with some analyzer)

### CODE REVIEW

File: src/KubernetesClient/IntOrStringConverter.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public override void Write(Utf8JsonWriter writer, IntstrIntOrString value, JsonS
         {
             var s = value?.Value;
 
-            if (int.TryParse(s, out var intv))
+            if (long.TryParse(s, out var intv))
```
Comment (WeihanLi) [inline src/KubernetesClient/IntOrStringConverter.cs:24]: From the code<https://github.com/kubernetes-client/csharp/blob/01b2aa832cfdc6f2c89153fe0123edc0a13b1bbf/src/KubernetesClient/IntstrIntOrString.cs#L11-L14>, we could see `long` is acceptable, so maybe we should use `long.TryParse`

Comment (tg123) [inline src/KubernetesClient/IntOrStringConverter.cs:24]: int str?

File: src/KubernetesClient/LineSeparatedHttpContent.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 namespace k8s
 {
-    internal class LineSeparatedHttpContent : HttpContent
+    internal sealed class LineSeparatedHttpContent : HttpContent
```
Comment (tg123) [inline src/KubernetesClient/LineSeparatedHttpContent.cs:9]: any reason to make internal class sealed?

Comment (WeihanLi) [reply src/KubernetesClient/LineSeparatedHttpContent.cs:9]: There's a ticket with some introduction https://github.com/dotnet/runtime/issues/49944

**Review by tg123 — APPROVED** [review]
/LGTM

**Pull request merged**